### PR TITLE
Ff internal documentation

### PR DESCRIPTION
**Pull request by alex-ozdemir**
Broadly:
* referencing parts of [OKTB23] where relevant
  * including in an overview of the files
* writing comments inside some procedures
* documenting the functionality of some subroutines
* improve some (internal) error messages

### CODE REVIEW

File: src/theory/ff/core.h [hunk 1]
```diff
@@ -65,39 +71,38 @@ class Tracer
...
-   */
+  /** Call this when s = spoly(p, q); */
   void sPoly(CoCoA::ConstRefRingElem p,
              CoCoA::ConstRefRingElem q,
              CoCoA::ConstRefRingElem s);
-  /**
-   * Call this when we start reducing p.
-   */
+  /** Tracing reduction p ->_q1 p1 ->_q2 p2 ->_q3 ... ->_qN -> r */
+  /** Call this when we start reducing p. */
```
[the remark this answers is not in the document]
Comment (alex-ozdemir) [reply src/theory/ff/core.h:81]: No, the first comment applies to the next 3 comments. Any recommendations for how to make that clear?

[the remark this answers is not in the document]
Comment (mpreiner) [reply src/theory/ff/core.h:81]: Ah ok, then maybe something like this makes it already clearer?
```
  /** Functions for tracing reduction p ->_q1 p1 ->_q2 p2 ->_q3 ... ->_qN -> r */

  /** Call this when we start reducing p. */
  void reductionStart(CoCoA::ConstRefRingElem p);
  /** Call this when there is a reduction on q. */
  void reductionStep(CoCoA::ConstRefRingElem q);
  /** Call this when we finish reducing with r. *
  void reductionEnd(CoCoA::ConstRefRingElem r)

  /** Internal helper functions */

  void addItem(const std::string&& item);
  void addDep(const std::string& parent, const std::string& child);
  ...
```

[the remark this answers is not in the document]
Comment (alex-ozdemir) [reply src/theory/ff/core.h]: [:+1:](https://github.com/cvc5/cvc5/pull/10051/commits/42fddbdaf8c0589089e97565d2f726e72e59a0e4)

**Review by mpreiner — APPROVED** [review]

**Comment by alex-ozdemir** [comment]
Whoops! Somehow I closed this. Re-opened, and IIRC, ready to merge.

**Pull request merged**